File: web/App_Code/LcEnum/UserType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace LcEnum
{
    public enum UserType : short
    {
        None = 0,
        Anonymous = 1,
        Client = 2,
        ServiceProfessional = 4,
        // All Members are Providers too,
        // so an option 'only member' does NOT exists
        // and its value gets reserved for use
        // grouped with the Provider (then, in binary 4 + 8 => 12)
        //OnlyMember = 8,
        Member = 12,
        Admin = 16,
        LoggedUser = 30,
        User = 31,
        System = 32
    }
}

File: app/source/js/enums/UserType.js
/**
 * Available types of users and special union values that match several types.
 * IMPORTANT: it matches same values on the server, any change should go in
 * sync
 * @enum {number}
  */
export default {
    none: 0,
    anonymous: 1,
    client: 2,
    serviceProfessional: 4,
    // All Members (member-only:8) are service professionals too: 4+8
    member: 12,
    admin: 16,
    // All users except anonymous and system:
    loggedUser: 30,
    // All users except system,
    user: 31,
    system: 32
};
